build/ci_utils/src/cache/goodie/graalvm.rs

```diff
@@ -54,27 +52,40 @@ impl Goodie for GraalVM {
 
     fn is_active(&self) -> BoxFuture<'static, Result<bool>> {
         let expected_graal_version = self.graal_version.clone();
-        let expected_java_language_version = self.java_version;
         async move {
             let found_version = find_graal_version().await?;
             ensure!(found_version == expected_graal_version, "GraalVM version mismatch. Expected {expected_graal_version}, found {found_version}.");
-
-            let found_java_version = Java.check_language_version().await?;
-            ensure!(
-                found_java_version == expected_java_language_version,
-                "Java language version mismatch. Expected {expected_java_language_version}, found {found_java_version}."
-            );
-
             Result::Ok(true)
         }
         .boxed()
     }
 
     fn activation_env_changes(&self, package_path: &Path) -> Result<Vec<crate::env::Modification>> {
-        let package_path = package_path.join(self.root_directory_name());
+        let dir_entries = package_path
+            .read_dir()
+            .context("Failed to read GraalVM cache directory")
+            .unwrap()
+            .collect_vec();
+        assert_eq!(
+            dir_entries.len(),
+            1,
+            "GraalVM cache directory should contain exactly one directory"
+        );
+        let graalvm_dir = match dir_entries.get(0).unwrap() {
+            Ok(dir_entry) => dir_entry,
+            Err(err) => bail!("Failed to read GraalVM cache directory: {}", err),
+        };
+        let dir_name_tmp = graalvm_dir.file_name();
+        let dir_name = dir_name_tmp.as_str();
+        assert!(dir_name.contains("graalvm"));
+        let graal_version_str = format!(
```

Comment:
Why not just `self.graal_version.to_string()` (unless we want to explicitly reject patch/prerelease pieces — but I don't see them in build.sbt?)
If we truly wanted to do this, I'd rather see this as a method in `VersionExt` (like `to_string_core` por sth), so the intent is clearly expressed.

Reply:
`graal_version_str` is only used for `assert!(dir_name.contains(&graal_version_str))`. The thing is that when you download the GraalVM distribution nowadays, and extract it, the directory name in the extracted folder has a weird name that includes the build metadata. And I have not found an easy way how to infer the directory name here. So I am just assuming that there is just one directory (as there has always been), and do some sanity checks on the name of the directory so that I am sure that the downloaded tarball contains the directory with graalvm.

Nevertheless, I added `to_string_core` method and its usages in 57c37ed3f08d7c0b07a14ba9aa6ccaeb8b07c0c5